git-discover/src/upwards/types.rs

```diff
@@ -156,4 +158,42 @@ mod tests {
 
         dir.close()
     }
+
+    #[test]
+    #[cfg(windows)]
+    fn parse_ceiling_dirs_from_environment_format() -> std::io::Result<()> {
+        use std::{fs, os::windows::fs::symlink_dir};
+
+        use super::*;
+
+        // Setup filesystem
+        let dir = tempfile::tempdir().expect("success creating temp dir");
+        let direct_path = dir.path().join("direct");
+        let symlink_path = dir.path().join("symlink");
+        fs::create_dir(&direct_path)?;
+        symlink_dir(&direct_path, &symlink_path)?;
+
+        // Parse & build ceiling dirs string
+        let symlink_str = symlink_path.to_str().expect("symlink path is valid utf8");
+        let ceiling_dir_string = format!("{};relative;;{}", symlink_str, symlink_str);
+        let ceiling_dirs = parse_ceiling_dirs(ceiling_dir_string.as_bytes());
+
+        assert_eq!(ceiling_dirs.len(), 2, "Relative path is discarded");
+        assert_eq!(
+            ceiling_dirs[0].to_string_lossy(),
+            symlink_path
+                .canonicalize()
+                .expect("symlink path exists")
+                .to_string_lossy()
+                .strip_prefix(r#"\\?\"#)
+                .expect("on Windows, the extended length path syntax is used"),
```

Comment:
I don't really like this, but [`std::fs::canonicalize`](https://doc.rust-lang.org/stable/std/fs/fn.canonicalize.html) returns a path prefixed with `\\?\`, which can't be removed by `PathBuf::strip_prefix`.